src/org/thoughtcrime/securesms/CreateProfileActivity.java

```diff
@@ -208,8 +218,8 @@ private void initializeResources() {
     this.statusView   = ViewUtil.findById(this, R.id.status_text);
 
     passwordAccountSettings.setOnClickListener(view -> {
-      Intent intent = new Intent(this, RegistrationActivity.class);
-      startActivity(intent);
+      boolean result = ScreenLockUtil.applyScreenLock(this, getString(R.string.pref_password_and_account_settings), getString(R.string.enter_system_secret_to_continue), ScreenLockUtil.REQUEST_CODE_CONFIRM_CREDENTIALS);
+      if (!result) { openRegistrationActivity(); }
```

Comment:
```suggestion
      if (!result) {
        openRegistrationActivity();
      }
```

although on-liners are used here and there, and sometimes, there are even reasons for it, i do not see them here.

but not a blocker, of course, feel free to leave as is :)